pkg/datapath/ipcache/listener.go

```diff
@@ -162,9 +162,13 @@ func (l *BPFListener) OnIPIdentityCacheChange(modType ipcache.CacheModification,
 			// with the hostIP so that this traffic can be guided
 			// to a tunnel endpoint destination.
 			nodeIPv4 := node.GetIPv4()
-			if ip4 := newHostIP.To4(); ip4 != nil && !ip4.Equal(nodeIPv4) {
-				copy(value.TunnelEndpoint[:], ip4)
+			// If the tunnel endpoint IPv4 address is manually set, don't change value.TunnelEndpoint
+			if tunnelIP := node.GetTunnelEndpointIPv4(); tunnelIP == nil {
+				if ip4 := newHostIP.To4(); ip4 != nil && !ip4.Equal(nodeIPv4) {
+					copy(value.TunnelEndpoint[:], ip4)
+				}
```

Comment:
Can you explain what you're trying to achieve here? I'm not sure I understand why tunnel should be treated differently.

Reply:
The reason here:
1. Originally, when there is NO tunnel-endpoint-manual-setting, the copy() action is executed only when the nodeIP(here nodeIPv4) is changed; if there's no change, then no copy() here;
2. When there is tunnel-endpoint-manual-setting, the value.TunnelEndpoint here is only setted by the manually selected IP address which matches the desired CIDR which should had been setted previously, so there's no change when the nodeIP is changed but now it's in tunnel-endpoint-manual-setting mode.